### PR TITLE
Fix integration of slack / serverless / aws

### PR DESCRIPTION
**Pull request by Startouf**
Rogntudjuuu !!!

### CODE REVIEW

File: messages/main.go [hunk 1]
```diff
@@ -22,51 +24,82 @@ type Response events.APIGatewayProxyResponse
...
+	handleSlackChallenge(eventsAPIEvent, body, resp)
+
+	if eventsAPIEvent.Type == slackevents.CallbackEvent || eventsAPIEvent.Type == slackevents.AppMention {
 		innerEvent := eventsAPIEvent.InnerEvent
+		log.Printf("Processing an event of inner data %s", innerEvent.Data)
 		switch ev := innerEvent.Data.(type) {
 		case *slackevents.AppMentionEvent:
-			api.PostMessage(ev.Channel, slack.MsgOptionText("Yes, hello.", false))
+			resp.StatusCode = 200
+			fmt.Printf("Reacting to app mention event from channel %s", ev.Channel)
```
Comment (gjgd) [inline messages/main.go]: any reasons why we use both `log.Printf` and `fmt.Printf` in this file?

File: messages/main.go [hunk 2]
```diff
@@ -22,51 +24,82 @@ type Response events.APIGatewayProxyResponse
...
 	return resp, nil
+}
 
+// Slack Challenge is used to register the URL in the slack API config interface
+// Should only be used once by slack when changing the events URL
+func handleSlackChallenge(eventsAPIEvent slackevents.EventsAPIEvent, body []byte, resp Response) {
+	if eventsAPIEvent.Type == slackevents.URLVerification {
+		buf := new(bytes.Buffer)
+		var r *slackevents.ChallengeResponse
+		err := json.Unmarshal(body, &r)
```
Comment (gjgd) [inline messages/main.go:97]: This will fail because r is null at this point

**Review by gjgd — DISMISSED** [review]
You need to update the makefile so that your messages/main function will be compiled before deploy:
https://github.com/gjgd/dont-slack-evil/blob/master/Makefile#L6

File: messages/main.go (hunk 2 above)
Comment (Startouf) [reply messages/main.go:97]: why is that ? Json.unmarshal unmarshals the content of the first variable in the second one, so it's good ?
(this code actually ran to validate the endpoint on the slack API website, so unless I messed something while refactoring, it should still work)

File: messages/main.go (hunk 1 above)
Comment (Startouf) [reply messages/main.go]: Good point. I think using log is a better practice, as I think it'll be easier to format the log later.

**Comment by Startouf** [comment]
@gjgd there is already this line in the Makefile `env GOOS=linux go build -ldflags="-s -w" -o bin/messages messages/main.go`

File: messages/main.go (hunk 2 above)
Comment (gjgd) [reply messages/main.go:97]: Oh yes, you're right mb

File: apphome/messages_awaiting_enlightenment.go [hunk 3]
```diff
@@ -41,12 +44,16 @@ func EnlightenmentMessages() []slack.Block {
 }
 
 func parseTestMessages() []Message {
-	jsonFile, err := os.Open("sample.json")
+	_, filename, _, _ := runtime.Caller(1)
+	file := path.Join(path.Dir(filename), "../data/sample.json")
```
Comment (gjgd) [inline apphome/messages_awaiting_enlightenment.go]: 👍  nice!

File: serverless.yml [hunk 4]
```diff
@@ -44,7 +45,9 @@ functions:
           method: post
     environment:
       SLACK_OAUTH_ACCESS_TOKEN: ${self:custom.secrets.SLACK_OAUTH_ACCESS_TOKEN}
+      SLACK_BOT_USER_OAUTH_ACCESS_TOKEN: ${self:custom.secrets.SLACK_BOT_USER_OAUTH_ACCESS_TOKEN}
```
Comment (gjgd) [inline serverless.yml:48]: update https://github.com/gjgd/dont-slack-evil/blob/0b8139ccb84663aca34ec6a19994ea374287eaed/example.secrets.dev.yml

File: apphome/messages_awaiting_enlightenment.go (hunk 3 above)
Comment (Startouf) [reply apphome/messages_awaiting_enlightenment.go]: Non en fait ça va tjrs pas... maintenant j'ai cassé le make deploy local. le serveur va chercher le fichier json au mauvais endroit `open /Users/Cyril/dev/dont-slack-evil/data/sample.json: no such file or directory`


:sigh:

**Review by gjgd — APPROVED** [review]

**Pull request merged**